### PR TITLE
[4.2] Move demo tasks to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Move the demo task plugin to service provider.

### Testing Instructions
Create different demo tasks like the memory stress test or sleep.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### CODE REVIEW

File: plugins/task/demotasks/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Task\DemoTasks\Extension\DemoTasks;
+use Joomla\Plugin\Task\SiteStatus\Extension\SiteStatus;
+use Joomla\Utilities\ArrayHelper;
```
Comment (heelc29) [inline plugins/task/demotasks/services/provider.php]:
```suggestion
```

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on ee60281e6a8c5c03e67e858e77f6853fe60c1ffb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38098">issues.joomla.org/tracker/joomla-cms/38098</a>.</sub>

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on ee60281e6a8c5c03e67e858e77f6853fe60c1ffb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38098">issues.joomla.org/tracker/joomla-cms/38098</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38098">issues.joomla.org/tracker/joomla-cms/38098</a>.</sub>

**Comment by richard67** [comment]
If this gets merged before 4.0.0-beta2, I will have to update the list of deleted files in script.php again. I would need to add file "/plugins/task/demotasks/demotasks.php" to the list of deleted files at line 6485.

I've prepared already a branch for that.

**Comment by alikon** [comment]
wait then ...
consider this too #38072

**Comment by richard67** [comment]
> wait then ... consider this too #38072

@alikon Sure, when it has RTC I will add the "plugins/task/checkfiles/checkfiles.php" file, too.

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**